api/v1alpha2/vspherecluster_conversion.go

```diff
@@ -43,6 +44,10 @@ func (src *VSphereCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 		return err
 	}
 
+	if restored.Spec.CloudProviderConfiguration.ProviderConfig.Cloud != nil {
+		dst.Spec.CloudProviderConfiguration.ProviderConfig.Cloud = restored.Spec.CloudProviderConfiguration.ProviderConfig.Cloud
```

Comment:
Should this just restore ExtraArgs, since ControllerImage exists in the previous version?